### PR TITLE
Use `version.major/minor/patch`

### PR DESCRIPTION
**Pull request by SeekingMeaning**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Same as #59485 but for non-GCC-related formulae

### CODE REVIEW

**Comment by SMillerDev** [comment]
You're changing install blocks. This is _not_ syntax-only

**Comment by SeekingMeaning** [comment]
Would manual verification be enough? I could go through each change and add a review comment with `brew irb` showing that they're the same

File: Formula/c7n.rb [hunk 1]
```diff
@@ -102,8 +102,7 @@ def install
 
   test do
     # trim last decimal point version to match semver returned from version command
-    cli_version = version.to_s.split(".")[0..2].join(".")
-    assert_match cli_version, shell_output("#{bin}/custodian version")
+    assert_match version.major_minor_patch.to_s, shell_output("#{bin}/custodian version")
```
Comment (SeekingMeaning) [inline Formula/c7n.rb:105]:
```
irb(main):001:0> Formula["c7n"].version.to_s.split(".")[0..2].join(".")
=> "0.9.4"
irb(main):002:0> Formula["c7n"].version.major_minor_patch.to_s
=> "0.9.4"
```

File: Formula/doctl.rb [hunk 2]
```diff
@@ -16,12 +16,11 @@ class Doctl < Formula
...
   def install
-    doctl_version = version.to_s.split(/\./)
     base_flag = "-X github.com/digitalocean/doctl"
     ldflags = %W[
-      #{base_flag}.Major=#{doctl_version[0]}
-      #{base_flag}.Minor=#{doctl_version[1]}
-      #{base_flag}.Patch=#{doctl_version[2]}
+      #{base_flag}.Major=#{version.major}
+      #{base_flag}.Minor=#{version.minor}
+      #{base_flag}.Patch=#{version.patch}
```
Comment (SeekingMeaning) [inline Formula/doctl.rb:23]:
```
irb(main):011:0> Formula["doctl"].version.to_s.split(/\./)
=> ["1", "46", "0"]
irb(main):012:0> Formula["doctl"].version.major
=> #<Version::NumericToken 1>
irb(main):013:0> Formula["doctl"].version.minor
=> #<Version::NumericToken 46>
irb(main):014:0> Formula["doctl"].version.patch
=> #<Version::NumericToken 0>
```

File: Formula/gitlab-runner.rb [hunk 3]
```diff
@@ -23,7 +23,7 @@ def install
     cd dir do
       proj = "gitlab.com/gitlab-org/gitlab-runner"
       commit = Utils.safe_popen_read("git", "rev-parse", "--short=8", "HEAD").chomp
-      branch = version.to_s.split(".")[0..1].join("-") + "-stable"
+      branch = "#{version.major}-#{version.minor}-stable"
```
Comment (SeekingMeaning) [inline Formula/gitlab-runner.rb:26]:
```
irb(main):005:0> Formula["gitlab-runner"].version.to_s.split(".")[0..1].join("-") + "-stable"
=> "13-3-stable"
irb(main):006:0> "#{Formula["gitlab-runner"].version.major}-#{Formula["gitlab-runner"].version.minor}-stable"
=> "13-3-stable"
```

File: Formula/go.rb [hunk 4]
```diff
@@ -8,7 +8,7 @@ class Go < Formula
     mirror "https://fossies.org/linux/misc/go1.15.src.tar.gz"
     sha256 "69438f7ed4f532154ffaf878f3dfd83747e7a00b70b3556eddabf7aaee28ac3a"
 
-    go_version = version.to_s.split(".")[0..1].join(".")
+    go_version = version.major_minor
```
Comment (SeekingMeaning) [inline Formula/go.rb:11]:
```
irb(main):007:0> Formula["go"].version.to_s.split(".")[0..1].join(".")
=> "1.15"
irb(main):008:0> Formula["go"].version.major_minor
=> #<Version:0x00007fbba10e1258 @version="1.15">
```

File: Formula/itk.rb [hunk 5]
```diff
@@ -80,7 +80,7 @@ def install
       }
     EOS
 
-    v = version.to_s.split(".")[0..1].join(".")
+    v = version.major_minor
```
Comment (SeekingMeaning) [inline Formula/itk.rb:83]:
```
irb(main):015:0> Formula["itk"].version.to_s.split(".")[0..1].join(".")
=> "5.1"
irb(main):016:0> Formula["itk"].version.major_minor
=> #<Version:0x00007fbba1110ad0 @version="5.1">
```

File: Formula/jmxtrans.rb [hunk 6]
```diff
@@ -27,10 +27,9 @@ def install
 
     cd "jmxtrans" do
       # Point JAR_FILE into Cellar where we've installed the jar file
-      vers = version.to_s.split("-").last
       inreplace "jmxtrans.sh", "$( cd \"$( dirname \"${BASH_SOURCE[0]}\" )/../lib\" "\
                 ">/dev/null && pwd )/jmxtrans-all.jar",
-                libexec/"target/jmxtrans-#{vers}-all.jar"
+                libexec/"target/jmxtrans-#{version}-all.jar"
```
Comment (SeekingMeaning) [inline Formula/jmxtrans.rb:32]:
```
irb(main):017:0> Formula["jmxtrans"].version.to_s.split("-").last
=> "271"
irb(main):018:0> Formula["jmxtrans"].version
=> #<Version::FromURL:0x00007fbb9e2c56a8 @version="271">
```

File: Formula/lbdb.rb [hunk 7]
```diff
@@ -21,7 +21,6 @@ def install
   end
 
   test do
-    ver = version.to_s.split(".")
-    assert_match "#{ver[0]}.#{ver[1]}", shell_output("#{bin}/lbdbq -v")
+    assert_match version.major_minor.to_s, shell_output("#{bin}/lbdbq -v")
```
Comment (SeekingMeaning) [inline Formula/lbdb.rb:24]:
```
irb(main):020:0> ver = Formula["lbdb"].version.to_s.split(".")
=> ["0", "48", "1"]
irb(main):021:0> "#{ver[0]}.#{ver[1]}"
=> "0.48"
irb(main):022:0> Formula["lbdb"].version.major_minor.to_s
=> "0.48"
```

File: Formula/mikutter.rb [hunk 8]
```diff
@@ -217,7 +217,7 @@ def install
     (lib/"mikutter").install "plugin"
     libexec.install Dir["*"]
 
-    ruby_series = Formula["ruby"].version.to_s.split(".")[0..1].join(".")
+    ruby_series = Formula["ruby"].installed_version.major_minor
```
Comment (SeekingMeaning) [inline Formula/mikutter.rb:220]:
```
irb(main):023:0> Formula["ruby"].version.to_s.split(".")[0..1].join(".")
=> "2.7"
irb(main):024:0> Formula["ruby"].installed_version.major_minor
=> #<Version:0x00007fbb9ea879b8 @version="2.7">
```

Note:

```patch
- version
+ installed_version
```

File: Formula/ncurses.rb [hunk 9]
```diff
@@ -33,7 +33,7 @@ def install
   end
 
   def make_libncurses_symlinks
-    major = version.to_s.split(".")[0]
+    major = version.major
```
Comment (SeekingMeaning) [inline Formula/ncurses.rb:36]:
```
irb(main):025:0> Formula["ncurses"].version.to_s.split(".")[0]
=> "6"
irb(main):026:0> Formula["ncurses"].version.major
=> #<Version::NumericToken 6>
```

File: Formula/php.rb [hunk 10]
```diff
@@ -286,14 +286,10 @@ def caveats
...
       The php.ini and php-fpm.ini file can be found in:
-          #{etc}/php/#{php_version}/
+          #{etc}/php/#{version.major_minor}/
     EOS
   end
 
-  def php_version
-    version.to_s.split(".")[0..1].join(".")
-  end
-
```
Comment (SeekingMeaning) [inline Formula/php.rb:296]:
```
irb(main):027:0> Formula["php"].version.to_s.split(".")[0..1].join(".")
=> "7.4"
irb(main):028:0> Formula["php"].version.major_minor
=> #<Version:0x00007fbba211def0 @version="7.4">
```

File: Formula/php@7.2.rb [hunk 11]
```diff
@@ -287,14 +287,10 @@ def caveats
...
       The php.ini and php-fpm.ini file can be found in:
-          #{etc}/php/#{php_version}/
+          #{etc}/php/#{version.major_minor}/
     EOS
   end
 
-  def php_version
-    version.to_s.split(".")[0..1].join(".")
-  end
-
```
Comment (SeekingMeaning) [inline Formula/php@7.2.rb:297]:
```
irb(main):029:0> Formula["php@7.2"].version.to_s.split(".")[0..1].join(".")
=> "7.2"
irb(main):030:0> Formula["php@7.2"].version.major_minor
=> #<Version:0x00007fbb9e287d58 @version="7.2">
```

File: Formula/php@7.3.rb [hunk 12]
```diff
@@ -292,14 +292,10 @@ def caveats
...
       The php.ini and php-fpm.ini file can be found in:
-          #{etc}/php/#{php_version}/
+          #{etc}/php/#{version.major_minor}/
     EOS
   end
 
-  def php_version
-    version.to_s.split(".")[0..1].join(".")
-  end
-
```
Comment (SeekingMeaning) [inline Formula/php@7.3.rb:302]:
```
irb(main):031:0> Formula["php@7.3"].version.to_s.split(".")[0..1].join(".")
=> "7.3"
irb(main):032:0> Formula["php@7.3"].version.major_minor
=> #<Version:0x00007fbba10c0ff8 @version="7.3">
```

File: Formula/postgresql.rb [hunk 13]
```diff
@@ -89,7 +89,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb --locale=C -E UTF-8 #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql.rb:92]:
```
irb(main):033:0> Formula["postgresql"].version.to_s.slice(/\d+/)
=> "12"
irb(main):034:0> Formula["postgresql"].version.major
=> #<Version::NumericToken 12>
```

File: Formula/postgresql@10.rb [hunk 14]
```diff
@@ -93,7 +93,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql@10.rb:96]:
```
irb(main):035:0> Formula["postgresql@10"].version.to_s.slice(/\d+/)
=> "10"
irb(main):036:0> Formula["postgresql@10"].version.major
=> #<Version::NumericToken 10>
```

File: Formula/postgresql@11.rb [hunk 15]
```diff
@@ -85,7 +85,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb --locale=C -E UTF-8 #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql@11.rb:88]:
```
irb(main):037:0> Formula["postgresql@11"].version.to_s.slice(/\d+/)
=> "11"
irb(main):038:0> Formula["postgresql@11"].version.major
=> #<Version::NumericToken 11>
```

File: Formula/postgresql@9.4.rb [hunk 16]
```diff
@@ -88,7 +88,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql@9.4.rb:91]:
```
irb(main):039:0> Formula["postgresql@9.4"].version.to_s.slice(/\d+/)
=> "9"
irb(main):040:0> Formula["postgresql@9.4"].version.major
=> #<Version::NumericToken 9>
```

File: Formula/postgresql@9.5.rb [hunk 17]
```diff
@@ -102,7 +102,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql@9.5.rb:105]:
```
irb(main):041:0> Formula["postgresql@9.5"].version.to_s.slice(/\d+/)
=> "9"
irb(main):042:0> Formula["postgresql@9.5"].version.major
=> #<Version::NumericToken 9>
```

File: Formula/postgresql@9.6.rb [hunk 18]
```diff
@@ -102,7 +102,7 @@ def caveats
       This formula has created a default database cluster with:
         initdb #{var}/postgres
       For more details, read:
-        https://www.postgresql.org/docs/#{version.to_s.slice(/\d+/)}/app-initdb.html
+        https://www.postgresql.org/docs/#{version.major}/app-initdb.html
```
Comment (SeekingMeaning) [inline Formula/postgresql@9.6.rb:105]:
```
irb(main):043:0> Formula["postgresql@9.6"].version.to_s.slice(/\d+/)
=> "9"
irb(main):044:0> Formula["postgresql@9.6"].version.major
=> #<Version::NumericToken 9>
```

**Comment by SMillerDev** [comment]
Then you're just doing CI locally.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**